### PR TITLE
refactor: add @packages/resolve-dist to isolate client/server code

### PR DESCRIPTION
**Pull request by tgriesser**
### Additional details

Begins setting up a cleaner separation between client & server code. The goal is to have each package serve one purpose - there should not be both node & client bundled code in the same folder. Adding `@packages/resolve-dist` consolidates the common pattern of finding the "dist" folder to a single location, which will make it easier to move the `client` dependencies around in the future.

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue or this PR been tagged with a release in ZenHub? 
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Review by lmiller1990 — DISMISSED** [review]
CI is ✅  . Is there anything particular you'd like to highlight here for reviewing? Seems we are just shuffling code around - no such change in functionality to scrutinize.

**Comment by tgriesser** [comment]
@lmiller1990 nope, just shuffling code around - just looking for the ✅ to merge

**Review by elevatebart — DISMISSED** [review]
LGTM

File: packages/server-ct/src/routes-ct.ts [hunk 1]
```diff
@@ -3,12 +3,12 @@ import { ErrorRequestHandler, Express } from 'express'
...
 import send from 'send'
 import { NetworkProxy } from '@packages/proxy'
-import { handle, serve, serveChunk } from '@packages/runner-ct'
+import { handle, serve, serveChunk } from './runner-ct'
 import xhrs from '@packages/server/lib/controllers/xhrs'
 import { SpecsStore } from '@packages/server/lib/specs-store'
-import staticPkg from '@packages/static'
 import { ProjectCt } from './project-ct'
 import { ProjectE2E } from '../../server/lib/project-e2e'
+import { getPathToDist } from '@packages/resolve-dist'
```
Comment (elevatebart) [inline packages/server-ct/src/routes-ct.ts:11]: Should `@packages/resolve-dist` be added to `packages/server-ct/package.json`?

Comment (tgriesser) [reply packages/server-ct/src/routes-ct.ts:11]: I don't think it should matter, but I suppose so to be consistent

**Comment by tgriesser** [comment]
@elevatebart and @lmiller1990 looks like the Percy tests related to - https://github.com/cypress-io/cypress/pull/16543 have you seen this happen elsewhere or is this somehow a regression with these changes?

**Comment by elevatebart** [comment]
@tgriesser these changes could be linked to the order in which the CSS is loaded. I will check it out after lunch, in an hour or so.

TTYT

**Comment by elevatebart** [comment]
I checked it out locally and checked that the CSS has not changed.

The moment those change appear is after viewport size changes. A slower/faster machine could make those changes more visible. But since I do not reproduce the 10px bottom margin locally, it looks like an environment related change.
Was the version of chrome on the container updated?

**Pull request merged**